### ISSUE TITLE
Action not completed Action cancelled by user / VM27:1 MetaMask - RPC Error: Error: Popup closed 

### ISSUE THREAD

**Issue by 0xGitGuy** [trezor/trezor-suite#12446]
**Describe the bug**
When you accidentally close trezor popup windows when signing for dapps like 1inch.io, the next attempt will cause the user to be unable to sign anymore transaction on Metamask. The trezor will be stuck and the only way to fix this is to deactivate metmask extension and enabling it again. So imagine a dapp action requiring 3 steps like (approve, sign, broadcast), pita. A non-hw address on metamask does not exhibit this issue. The last time Model One firmware had an update was March 2023 (1 years+ ago).  Does Trezor deprecate support for older devices over time? (this sucks though for storing funds)

**Firmware version and revision**
Brave (1.65.133 (May 14, 2024)), Windows 10, Metamask (11.15.4) , Trezor Model One (1.12.1)

**Desktop/smartphone setup (please complete the following information):**
as above

**To Reproduce**
1. attempt to sign a transaction at 1inch.io
2. once trezor model one hardware shows some text to sign, click Confirm on metamask once first
3. once you see the popup.html, close the trezor pop up window on your pc
4. now try the same transaction for 1inch.io again
5. you either get a infinite trezor loading screen or it shows you an error

**Expected behavior**
Type 4 signing should be able to sign transaction without resetting Metamask on the second transaction attempt

**Screenshots**
Dapp console error shows:
VM27:1 MetaMask - RPC Error: Error: Popup closed 

Trezor popup.html windows error shows:
Action not completed Action cancelled by user. 
(Eternally spinning the loading image)

**Additional context**
I did have a way to trigger the Model One again for the second transaction but it's way too cumbersome and not normal UI

**Comment by Hannsek** [comment]
> Does Trezor deprecate support for older devices over time?

The support for this device was not deprecated. There was no need to release new fw for Model 1.


> the next attempt will cause the user to be unable to sign anymore transaction on Metamask.

How does it look You should trigger signing the transaction in MM again and Connect popup should popup again.

**Comment by 0xGitGuy** [comment]
> > the next attempt will cause the user to be unable to sign anymore transaction on Metamask.
> 
> How does it look You should trigger signing the transaction in MM again and Connect popup should popup again.

Just for clarification, yes, this is correct. It should cancel the previous transaction and push the newly attempted transaction to Model One.

**Comment by Hannsek** [comment]
So what happens when you click on sign in MM again?

**Comment by 0xGitGuy** [comment]
> So what happens when you click on sign in MM again?

Model One screen gets stuck from the previous transaction. Clicking either buttons or triggering another transaction on Metamask doesn't do anything to change Model One screen. I need to disable metamask. In most cases,

**Comment by Hannsek** [comment]
Ok, we'll investigate. cc @trezor/qa

**Comment by bosomt** [comment]
**QA OK**

Tried to send Eth transaction via https://app.1inch.io/#/1/simple/swap/ETH
Closed popup multiple times https://connect.trezor.io/9/popup.html
It was auto opened till  `Transaction rejected` was triggered
After that i was able to open popup again without any kind of problem.

**Comment by bosomt** [comment]
- Tried to rexplicate by 1inch + MeaMask + https://metamask.github.io/test-dapp/
- got when trying to sign v4 data not sure if related  see video https://youtu.be/cU7vdPauERc
- Error in testdapp: `Result: Error: Keyring Controller signTypedMessage: Error: Device is used in another window`
- Error in connect popup before confirming passphrase 
<img width="1391" alt="image" src="https://github.com/trezor/trezor-suite/assets/31506317/633edd88-7649-4153-9e43-a9f7c38bb517">

**Comment by 0xGitGuy** [comment]
> * Tried to rexplicate by 1inch + MeaMask + https://metamask.github.io/test-dapp/
> * got when trying to sign v4 data not sure if related  see video https://youtu.be/cU7vdPauERc
> * Error in testdapp: `Result: Error: Keyring Controller signTypedMessage: Error: Device is used in another window`
> * Error in connect popup before confirming passphrase
> 
> <img alt="image" width="1391" src="https://private-user-images.githubusercontent.com/31506317/332844135-633edd88-7649-4153-9e43-a9f7c38bb517.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzOTQ1NzksIm5iZiI6MTcxNjM5NDI3OSwicGF0aCI6Ii8zMTUwNjMxNy8zMzI4NDQxMzUtNjMzZWRkODgtNzY0OS00MTUzLTllNDMtYTlmN2MzOGJiNTE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDE2MTExOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxODM5NmQ1NjhhMTM3MjNmYjE3ZWJmM2IyZDFlZjFmMTdhOGY3YjUxNzE3MjFkNzVkOTBiMTExYzRiOTc3NDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.km7P8iCw8kHTjiDiKI8yCLya7KdKVY4uHPkdIrL61xU">

oh yes, this is it! Same console errors as I got. Boy, I wished metamask and trezor don't do this to me every now and then when it force updates. 

Did you find a temp way to quickly recover from the bug without disable/enable metamask and go through the password/allow phases?

**Comment by mroz22** [comment]
> When you accidentally close trezor popup windows when signing

Yes, I confirm this is where the problem starts. In the best case scenario, trezor-connect should successfully "release" device when connect-popup is closed. There are couple of modes of how trezor-connect can be used, and we are actually fixing this problem in one of the scenarios (different from yours) [here](https://github.com/trezor/trezor-suite/pull/12198). 

Could you @martykan please check if core in iframe mode actually releases device if it learns about the fact that popup was closed? I think it should but apparently it does not.

As I stated above, this is the best case scenario, so it is not guaranteed that it will succeed. So there should be a fallback in popup, a button "use device here". This button is there but for some reason popup closes sooner before this button is rendered. 

Anyway thanks for reporting, we are already investigating.

**Comment by bosomt** [comment]
Sorry im not able to reproduce this issue today :)))
Tried to run new and old bridge with no luck
Even opened Rabby wallet in same browser so it will interupt connection but i was able to sign v4 data 10 of 10 times...

**Comment by martykan** [comment]
Closing as stale, haven't seen reports of this recently
